github/__init__.py

```diff
@@ -227,12 +228,23 @@ async def repositories(self) -> list[Repository]:
                 except KeyError:
                     return result
 
-    async def get_check_suites(self, ref: str) -> list:
-        # https://docs.github.com/en/rest/reference/checks#list-check-suites-for-a-git-reference
-        response = await self.get(f"/repos/{self.repo_path}/commits/{ref}/check-suites")
-        GithubError.raise_from_message(response)
+    async def get_check_runs(self, ref: str) -> list:
+        """
+        Returns all check runs for the commit ref
+        """
 
-        if "check_suites" not in response:
-            raise KeyError
+        async def _get_check_runs(per_page: int, page: int) -> tuple[list, int]:
+            # https://docs.github.com/en/rest/reference/checks#list-check-runs-for-a-git-reference
+            response = await self.get(
+                f"/repos/{self.repo_path}/commits/{ref}/check-runs?per_page={per_page}&page={page}",
+            )
+            GithubError.raise_from_message(response)
+            return response["check_runs"], response["total_count"]
+
+        check_runs, total_count = await _get_check_runs(per_page=100, page=1)
+        for runs, _ in await asyncio.gather(
+            *(_get_check_runs(per_page=100, page=page + 2) for page in range(total_count // 100)),
```

Comment:
I think the new code is not 100% correct for the case `total_count=100`. It loop over `range(1)` which will then fetch `page=2`. But all 100 entries fit on page 1, so there is no need to fetch page=2 (maybe it is even an error, I don't know).

Reply:
There is no error, GitHub return an empty list. But you're right. I need to fix the code according to that edge case.